examples/hello_world/client/rust/src/main.rs

```diff
@@ -87,5 +88,34 @@ async fn main() -> anyhow::Result<()> {
     // Join all the tasks. NB: this is where the tasks are being run!
     futures::future::join_all(responses).await;
 
+    let req = Request::new(HelloRequest {
+        greeting: "WORLDS".to_string(),
+    });
+    let responses: Vec<HelloResponse> = client
+        .lots_of_replies(req)
+        .await
+        .expect("could not receive streaming response")
+        .into_inner()
+        .collect::<Vec<Result<HelloResponse, tonic::Status>>>()
+        .await
+        .into_iter()
+        .filter_map(|v| v.ok())
+        .collect();
+    // Make sure that the translated response was received.
+    assert_eq!(
+        vec![
+            HelloResponse {
+                reply: "HELLO WORLDS!".to_string()
+            },
+            HelloResponse {
+                reply: "BONJOUR MONDES!".to_string()
+            },
+            HelloResponse {
+                reply: "HELLO AGAIN WORLDS!".to_string()
+            }
+        ],
+        responses
+    );
```

Comment:
I added this assertion here because otherwise I was not able to notice any failure when I incorrectly instantiated the translator module, so that part of the application was never properly exercised.

This does not work for the C++ application though, since that only ever returns a single hardcoded response every time:

https://github.com/project-oak/oak/blob/aa8c63d40fc3022d49238eb6aead0ff9537b015a/examples/hello_world/module/cpp/hello_world.cc#L25-L39

but also the wider question is whether we should really have separate examples from tests; lately it seems that tests are basically reinventing most of the logic around compiling modules, running oak application, connecting over gRPC etc. which is pretty close to what runner does with examples. I think we should start thinking how to unify the two worlds, or have clearer guidelines on when something is a test or an example.

WDYT?

Reply:
Yeah, the `hello_world` example client was previously deliberately lax, so that it would successfully complete for any of:
 - connecting to Rust module with translator attached
 - connecting to Rust module without translator attached
 - connecting to C++ module.

My general feeling is that the primary purpose of the (non-`abitest`) examples is to demonstrate use of the platform, and so the standalone clients should generally *not* be asserting expected responses from the Oak application being exercised.  If there's particular behaviour that needs testing, then:

 - if it's application behaviour, the test should live in `examples/xyzzy/module/rust/test/integration_test.rs` (and can assert expected responses)
 - if it's Runtime behaviour, the test should live in `abitest`

For this particular case, maybe the we should just make the `translator` non-optional for `hello_world`/Rust?

Reply:
You are right, I moved these to the integration tests now.